plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/GenericStatsMetricPublisher.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.s3;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Generic stats of repository-s3 plugin.
+ */
+public class GenericStatsMetricPublisher {
+
+    private final AtomicLong normalPriorityQSize = new AtomicLong();
+    private final AtomicInteger normalPriorityPermits = new AtomicInteger();
+    private final AtomicLong lowPriorityQSize = new AtomicLong();
+    private final AtomicInteger lowPriorityPermits = new AtomicInteger();
+
+    public void updateNormalPriorityQSize(long qSize, boolean increment) {
+        if (increment) {
+            normalPriorityQSize.addAndGet(qSize);
+        } else {
+            normalPriorityQSize.addAndGet(-qSize);
+        }
+    }
+
+    public void updateLowPriorityQSize(long qSize, boolean increment) {
+        if (increment) {
+            lowPriorityQSize.addAndGet(qSize);
+        } else {
+            lowPriorityQSize.addAndGet(-qSize);
+        }
```

Comment:
Why not just pass a qSize thats either positive or negative rather than a supporting boolean increment

Reply:
Yes, makes sense. Done.